app/assets/javascripts/dynamic_fields.js.coffee

```diff
@@ -48,13 +48,27 @@ $ ->
           new_input_group.find('.twitter-typeahead').remove()
           initialize_typeahead(new_input)
         else if current_input_group.find('.dropdown-menu').size()
-          dropdown_default = current_input_group.find('.dropdown-menu li:first a').text()
-          new_input_group.find('.dropdown-toggle span').first().text(dropdown_default)
-          new_input_group.find('input[type="hidden"]').val(dropdown_default)
+          dropdown_default_label = current_input_group.find('.dropdown-menu li:first a').text()
+          dropdown_default_value = current_input_group.find('.dropdown-menu li:first span').text()
+          new_input_group.find('.dropdown-toggle span').first().text(dropdown_default_label)
+          new_input_group.find('input[type="hidden"]').val(dropdown_default_value)
         current_input_group.find('.input-group-btn').has('.add-dynamic-field').remove()
         current_input_group.append(remove_button_html)
-        current_input_group.after(new_input_group)
+        current_textarea = $('#'+current_input_group.data('textarea'))
+        if current_textarea.size()
+          new_textarea = current_textarea.clone()
+          new_textarea.val('')
+          idArray = new_textarea.attr('id').split('_')
+          idArray.push(parseInt(idArray.pop()) + 1)
+          new_textarea.attr('id', idArray.join('_'))
+          new_input_group.attr('data-textarea', new_textarea.attr('id'))
+          current_textarea.after(new_input_group)
+          new_input_group.after(new_textarea)
+        else
+          current_input_group.after(new_input_group)
         
       $(document).on 'click', '.remove-dynamic-field', (e) ->
         e.preventDefault()
-        $(this).closest('.input-group').remove()
+        current_input_group = $(this).closest('.input-group')
+        $('#'+current_input_group.data('textarea')).remove()
```

Comment:
This line looks like it will try to remove "#undefined" if there is no textarea associated with the input group.  I think we should protect this line so it will only be run if there is a textarea defined.